crates/uv/tests/it/pip_install.rs

```diff
@@ -5425,19 +5425,19 @@ fn install_package_basic_auth_from_keyring_wrong_username() {
         .arg("subprocess")
         .arg("--strict")
         .env(EnvVars::KEYRING_TEST_CREDENTIALS, r#"{"pypi-proxy.fly.dev": {"other": "heron"}}"#)
-        .env(EnvVars::PATH, venv_bin_path(&context.venv)), @r###"
+        .env(EnvVars::PATH, venv_bin_path(&context.venv)), @r"
     success: false
     exit_code: 1
     ----- stdout -----
 
     ----- stderr -----
-    Request for public@https://pypi-proxy.fly.dev/basic-auth/simple/anyio/
-    Request for public@pypi-proxy.fly.dev
+    Keyring request for public@https://pypi-proxy.fly.dev/basic-auth
+    Keyring request for public@pypi-proxy.fly.dev
```

Comment:
Sorry if I'm just missing this, but... is there a test case here where the keyring has credentials for the index level and we _don't_ fallback to the realm request?

Reply:
Similarly, is there a test case where the keyring has credentials for the full URL but not at the index or realm level?

(These would be achieved by tweaking `EnvVars::KEYRING_TEST_CREDENTIALS, r#"{"pypi-proxy.fly.dev": {"other": "heron"}}"`)

Reply:
> Sorry if I'm just missing this, but... is there a test case here where the keyring has credentials for the index level and we _don't_ fallback to the realm request?

That test is in the follow-on PR #12717 (which we can merge into this one).


> Similarly, is there a test case where the keyring has credentials for the full URL but not at the index or realm level?
> 
> (These would be achieved by tweaking `EnvVars::KEYRING_TEST_CREDENTIALS, r#"{"pypi-proxy.fly.dev": {"other": "heron"}}"`)

I don't have this test, so I can add it.